src/App.js

```diff
@@ -23,49 +25,107 @@ const generateSquares = () => {
   }
 
   return squares;
-}
+};
 
 const App = () => {
-
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
+
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState(null);
+
+  const onClickCallback = (id) => {
+    setSquares((squares) =>{
```

Comment:
There's no check for whether the game is over, so it's possible to continue placing pieces after the game is "over." In fact, the winner can change if we keep placing pieces, and a winning line is found earlier in the check for winner logic than the original winning line. This isn't the behavior we want.